sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/ResourceExtensionsTests.cs

```diff
@@ -8,236 +8,312 @@
+            Assert.Equal("my-instance", metricsData?.Properties[SemanticConventions.AttributeServiceInstance]);
+            Assert.Equal("bar", metricsData?.Properties["foo"]);
         }
+    }
 
-        [Theory]
-        [InlineData(null)]
-        [InlineData(InstrumentationKey)]
-        public void MetricTelemetryHasAllResourceAttributes(string? instrumentationKey)
+    [Fact]
+    public void ResourceWithKubernetesAttributes()
+    {
+        // Arrange
+        var testAttributes = new Dictionary<string, object>
         {
-            var testAttributes = new Dictionary<string, object>
-            {
-                {"service.name", "my-service" },
-                {"service.namespace", "my-namespace" },
-                {"service.instance.id", "my-instance" },
-                { "foo", "bar" }
-            };
-
-            var resource = ResourceBuilder.CreateEmpty().AddAttributes(testAttributes).Build();
-            var azMonResource = resource.CreateAzureMonitorResource(instrumentationKey);
+            {SemanticConventions.AttributeK8sDeployment, "my-deployment" },
+            {SemanticConventions.AttributeK8sPod, "my-pod" },
+        };
 
-            Assert.Equal(instrumentationKey != null, azMonResource?.MetricTelemetry != null);
+        var resource = ResourceBuilder.CreateEmpty().AddAttributes(testAttributes).Build();
+        var azMonResource = resource.CreateAzureMonitorResource();
 
-            if (instrumentationKey != null)
-            {
-                Assert.Equal("Metric", azMonResource!.MetricTelemetry!.Name);
-                Assert.Equal(3, azMonResource.MetricTelemetry.Tags.Count);
-                Assert.NotNull(azMonResource.MetricTelemetry.Data);
+        // Assert
+        Assert.Equal("my-deployment", azMonResource?.RoleName);
+        Assert.Equal("my-pod", azMonResource?.RoleInstance);
+    }
 
-                var monitorBase = azMonResource.MetricTelemetry.Data;
-                var metricsData = monitorBase.BaseData as MetricsData;
+    [Fact]
+    public void ResourceWithKubernetesAndServiceAttributes()
+    {
+        // Arrange
+        var testAttributes = new Dictionary<string, object>
+        {
+            {SemanticConventions.AttributeServiceName, "my-service" },
+            {SemanticConventions.AttributeServiceInstance, "my-instance" },
+            {SemanticConventions.AttributeK8sDeployment, "my-deployment" },
+            {SemanticConventions.AttributeK8sPod, "my-pod" },
+        };
+
+        var resource = ResourceBuilder.CreateEmpty().AddAttributes(testAttributes).Build();
+        var azMonResource = resource.CreateAzureMonitorResource();
+
+        // Assert
+        Assert.Equal("my-service", azMonResource?.RoleName);
+        Assert.Equal("my-instance", azMonResource?.RoleInstance);
+    }
 
-                Assert.NotNull(metricsData?.Metrics);
+    [Fact]
+    public void ResourceWithKubernetesServiceAndCustomAttributes()
```

Comment:
What is the difference between this one and `ResourceWithKubernetesAndServiceAttributes`? Do we need this one?

Reply:
Only difference is a custom attribute. Wanted to show custom attribute does not impact the result. If you feel it is not needed, we could remove it.

Reply:
I think we can keep this one and remove `ResourceWithKubernetesAndServiceAttributes` as the kubernetes/custom attributes will be treated the same(i.e. will be dropped) in presence of service.name/instance for setting cloud.role and cloud.role.instance.